rust/datafusion/src/logical_plan/builder.rs

```diff
@@ -178,6 +178,29 @@ impl LogicalPlanBuilder {
         }))
     }
 
+    /// Apply a union
+    pub fn union(&self, plan: LogicalPlan) -> Result<Self> {
+        let schema = self.plan.schema();
+
+        if plan.schema() != schema {
+            return Err(DataFusionError::Plan(
+                "Schema's for union should be the same ".to_string(),
+            ));
+        }
+        // Add plan to existing union if possible
+        let mut inputs = match &self.plan {
```

Comment:
I think this case should have its own test too (or remove this optimization) 

I am thinking something like 

```
builder.union(p1)
.union(p2)
```
and then verify there is only a single `LogicalPlan::Union` in the output